### PR TITLE
add processor unittest

### PR DESCRIPTION
**Pull request by cgDeepLearn**
add processor unitttest

### Which issue(s) this PR fixes:
Fixes https://github.com/seata/seata-go/issues/27

### Special notes for your reviewer:
 client rmBranch commit/roolback test need a server
### Does this PR introduce a user-facing change?:

### CODE REVIEW

File: pkg/remoting/processor/client/client_heart_beat_processor_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	"github.com/seata/seata-go/pkg/protocol/codec"
+	"github.com/seata/seata-go/pkg/protocol/message"
+)
+
+func TestClientHeartBeatProcessor(t *testing.T) {
+	// testcases
+	var tests = []struct {
+		name string // testcase name
+
```
Comment (luky116) [inline pkg/remoting/processor/client/client_heart_beat_processor_test.go:32]: 空行可以去了

File: pkg/remoting/processor/client/client_on_response_processor_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	"context"
+	"testing"
+
+	"github.com/seata/seata-go/pkg/protocol/codec"
+	"github.com/seata/seata-go/pkg/protocol/message"
+)
+
+func TestClientOnResponseProcessor(t *testing.T) {
+	// testcases
+	var tests = []struct {
```
Comment (luky116) [inline pkg/remoting/processor/client/client_on_response_processor_test.go:30]: 可以把 getty2.GetGettyRemotingInstance().GetMergedMessage 和 getty2.GetGettyRemotingInstance().RemoveMergedMessageFuture 做个mock来测试，这样能测试的覆盖率能更广

File: pkg/remoting/processor/client/rm_branch_commit_processor_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+			wantErr: true, // need dail to server, so err accured
+		},
+	}
+
+	var ctx context.Context
+	var rbcProcessor rmBranchCommitProcessor
+
+	rm.GetResourceManagerInstance().RegisterResourceManager(tcc.GetTCCResourceManagerInstance())
+	// tccResource := tcc.TCCResource{
```
Comment (luky116) [inline pkg/remoting/processor/client/rm_branch_commit_processor_test.go:72]: 注释代码可以删了

File: pkg/remoting/processor/client/rm_branch_rollback_processor_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			 wantErr: true, // need dail to server, so err accured
+		 },
+	 }
+ 
+	 var ctx context.Context
+	 var rbrProcessor rmBranchRollbackProcessor
+ 
+	 rm.GetResourceManagerInstance().RegisterResourceManager(tcc.GetTCCResourceManagerInstance())
+	 // tccResource := tcc.TCCResource{
+	 // 	ActionName: "1232324",
```
Comment (luky116) [inline pkg/remoting/processor/client/rm_branch_rollback_processor_test.go:73]: 注释代码可以删了

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**